src/core/src/GslMatrix.C

```diff
@@ -692,15 +692,17 @@ GslMatrix::filterLargeValues(double thresholdValue)
 GslMatrix
 GslMatrix::transpose() const
 {
-  unsigned int nRows = this->numRowsLocal();
+  unsigned int nRows = this->numRowsGlobal();
   unsigned int nCols = this->numCols();
 
-  queso_require_equal_to_msg(nRows, nCols, "routine works only for square matrices");
+  const MpiComm & comm = this->map().Comm();
+  Map serial_map(nCols, 0, comm);
+
+  GslMatrix mat(m_env,serial_map,nRows);
 
-  GslMatrix mat(m_env,m_map,nCols);
   for (unsigned int row = 0; row < nRows; ++row) {
     for (unsigned int col = 0; col < nCols; ++col) {
-      mat(row,col) = (*this)(col,row);
+      mat(col,row) = (*this)(row,col);
```

Comment:
The other lines were to get transposes working with square matrices.  This line was changed to keep the rectangular dimensions consistent with the definitions above.

I didn't think about the possibility of performance change.  I don't think there's any clearly-winning way to do a transpose, since if you're doing a nice stride on one matrix you're doing a poor stride on the other, but we could swap the loop order if I'm wrong.